commands/src/main/java/org/wildfly/extras/creaper/commands/messaging/AbstractTransportConfigAddCommand.java

```diff
@@ -0,0 +1,230 @@
+package org.wildfly.extras.creaper.commands.messaging;
+
+import org.wildfly.extras.creaper.core.CommandFailedException;
+import org.wildfly.extras.creaper.core.online.OnlineCommand;
+import org.wildfly.extras.creaper.core.online.OnlineCommandContext;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.OperationException;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.Values;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * An abstract class which implements common properties of connector and acceptor in messaging connections.
+ */
+abstract class AbstractTransportConfigAddCommand implements OnlineCommand {
+
+    protected TransportConfigType type;
+    protected String name;
+    protected String serverName;
+    protected Map<String, String> params;
+    protected boolean replaceExisting;
+
+    //in-vm
+    protected Integer serverId;
+
+    //generic and remote
+    protected String socketBinding;
+
+    //remote
+    protected String factoryClass;
```

Comment:
Not critical, really. You could create a single `private` constructor that would take all the parameters and set the fields; all the other `protected` constructors would just delegate to this. But as I said, I don't insist on this; I just like the commands to be immutable.